api/src/main/java/uk/gov/dhsc/htbhf/claimant/model/UpdatableClaimantFields.java

```diff
@@ -1,16 +1,107 @@
 package uk.gov.dhsc.htbhf.claimant.model;
 
-import lombok.AllArgsConstructor;
 import lombok.Getter;
+import uk.gov.dhsc.htbhf.claimant.entity.Address;
+import uk.gov.dhsc.htbhf.claimant.entity.Claimant;
+
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.util.*;
+import java.util.stream.Collectors;
 
 /**
  * Enumeration of all the fields on a Claimant that may be updated by the claimant.
  */
-@AllArgsConstructor
-@Getter
 public enum UpdatableClaimantFields {
 
-    EXPECTED_DELIVERY_DATE("expectedDeliveryDate");
+    EXPECTED_DELIVERY_DATE("expectedDeliveryDate"),
+    ADDRESS("address") {
+
+        @Override
+        boolean valueIsDifferent(Claimant originalClaimant, Claimant newClaimant) throws InvocationTargetException, IllegalAccessException {
+            return hasAddressChanged(originalClaimant, newClaimant);
+        }
+
+        @Override
+        void updateOriginal(Claimant originalClaimant, Claimant newClaimant) {
+            updateAddress(originalClaimant, newClaimant);
+        }
+    };
 
+    @Getter
     private final String fieldName;
+    private final Method setter;
+    private final Method getter;
+
+    /**
+     * Constructor that accepts the field name and uses it to obtain the getter and setter methods.
+     * @param fieldName the name of the field.
+     */
+    UpdatableClaimantFields(String fieldName) {
+        this.fieldName = fieldName;
+        Map<String, Method> methodMap = Arrays.stream(Claimant.class.getDeclaredMethods()).collect(Collectors.toMap(
```

Comment:
I tried to make this map static, but it couldn't then be referred to from an enum constructor